arangod/Aql/IResearchViewExecutor.cpp

```diff
@@ -680,59 +672,140 @@ bool IResearchViewExecutorBase<Impl, Traits>::writeRow(ReadContext& ctx,
                                                        LocalDocumentId const& documentId,
                                                        LogicalCollection const& collection) {
   TRI_ASSERT(documentId.isSet());
-  bool writeDocOk;
-  if constexpr (Traits::MaterializeType == MaterializeType::Materialized) {
+  if constexpr (Traits::MaterializeType == MaterializeType::Materialize) {
     // read document from underlying storage engine, if we got an id
-    writeDocOk = collection.readDocumentWithCallback(infos().getQuery().trx(), documentId, ctx.callback);
-  } else {
+    if (!collection.readDocumentWithCallback(infos().getQuery().trx(), documentId, ctx.callback)) {
+      return false;
+    }
+  } else if ((Traits::MaterializeType & MaterializeType::LateMaterialize) == MaterializeType::LateMaterialize) {
     // no need to look into collection. Somebody down the stream will do materialization. Just emit LocalDocumentIds
-    writeDocOk = writeLocalDocumentId(ctx, documentId, collection);
+    if (!writeLocalDocumentId(ctx, documentId, collection)) {
+      return false;
+    }
   }
-  if (writeDocOk) {
-    // in the ordered case we have to write scores as well as a document
-    if constexpr (Traits::Ordered) {
-      // scorer register are placed right before the document output register
-      RegisterId scoreReg = infos().getFirstScoreRegister();
-      for (auto& it : _indexReadBuffer.getScores(bufferEntry)) {
-        TRI_ASSERT(infos().isScoreReg(scoreReg));
-        bool mustDestroy = false;
-        AqlValueGuard guard{it, mustDestroy};
-        ctx.outputRow.moveValueInto(scoreReg, ctx.inputRow, guard);
-        ++scoreReg;
+  if constexpr ((Traits::MaterializeType & MaterializeType::UseStoredValues) == MaterializeType::UseStoredValues) {
+    auto const& columnsFieldsRegs = infos().getOutNonMaterializedViewRegs();
+    TRI_ASSERT(!columnsFieldsRegs.empty());
+    auto max = (--columnsFieldsRegs.cend())->first;
```

Comment:
Please consider extracting acces to max and columnsFieldsRegs into separate (possibly inline) function